src/core/node.ts

```diff
@@ -109,12 +109,12 @@ export abstract class Node {
                 localizedPatch = patch
             else
                 localizedPatch = extend({}, patch, {
-                    path: getRelativePath(this, source)
+                    path: getRelativePath(this, source) + patch.path
```

Comment:
This is necessary because ArrayNode emits patches with indices (not existing ones for example /rows/0) that should bubble up to the top. I changed the string to be concatenated and the current node to be passed up as the source.